packages/server/src/models/StreamSplitter.ts

```diff
@@ -117,13 +119,23 @@ export class StreamSplitter extends stream.Writable {
     this.part += 1
   }
 
+  async emitEvent<T>(name: string, payload: T) {
```

Comment:
This function should be named differently because it doesn't emit events in an asynchronous fashion like we are used from EventEmitter. Instead, it sequentially invokes callbacks, which is different.

Reply:
the standard EventEmitter emits events **synchronously** this function does the same thing but it allows using async/await to have a similar behavior to the standard EventEmitter

Reply:
In essence the standard EventEmitter loops sequentially all events as my implementation and the handler blocks in a synchronous fashion, however, if you do async operations it won't await by default

Reply:
You are right, thanks for correct me! Then the name is fine.